x/staging/posts/simulation/operations_reports.go

```diff
@@ -86,14 +88,23 @@ func sendMsgReportPost(
 }
 
 func randomReportPostFields(
-	r *rand.Rand, ctx sdk.Context, accs []simtypes.Account, ak authkeeper.AccountKeeper, pk postskeeper.Keeper,
+	r *rand.Rand, ctx sdk.Context, accs []simtypes.Account, ak authkeeper.AccountKeeper, pk postskeeper.Keeper, sk subspaceskeeper.Keeper,
 ) (*ReportsData, bool) {
 	posts := pk.GetPosts(ctx)
 	if posts == nil {
 		return nil, true
 	}
 
+	subspace, _ := subspacessims.RandomSubspace(r, sk.GetAllSubspaces(ctx))
+
 	reportsData := RandomReportsData(r, posts, accs)
+	post, found := pk.GetPost(ctx, reportsData.PostID)
+	if !found {
+		return nil, true
+	}
+	post.Subspace = subspace.ID
+	pk.SavePost(ctx, post)
```

Comment:
I don't get it why we need to continuously change the post subspace before performing any operations on it. What happens if we remove this?

Reply:
To ensure we won't have any errors from a non-existent subspace, do you think we can remove this?

Reply:
@bragaz As far as I remember, posts are created in two occasions: 

1. Inside the genesis 
2. Using `MsgCreatePost` 

What we can do is making sure that while initializing them inside those places, we subspace id is taken randomly from the slice of existing subspaces. This would make sure that we never have to check for such errors. 

So, in summary, we should: 

1. Generate a random slice of genesis subspaces. 
2. Generate a random slice of genesis posts, each one having subspace id randomly taken from the genesis subspaces. 
3. When we perform a `MsgCreatePost`, we get all the existing subspaces, extract one random subspace and get the id. Then we use that ID as the post subspace id